src/algebraic_geometry/open_immersion.lean

```diff
@@ -1205,6 +1205,17 @@ def open_cover.pushforward_iso {X Y : Scheme} (𝒰 : open_cover X)
   (λ _, iso.refl _)
   (λ _, (category.id_comp _).symm)
 
+/-- Adding an open immersion into an open cover gives another open cover. -/
+@[simps]
+def open_cover.add {X : Scheme} (𝒰 : X.open_cover) {Y : Scheme} (f : Y ⟶ X)
+  [is_open_immersion f] : X.open_cover :=
+{ J := option 𝒰.J,
+  obj := λ i, option.rec Y 𝒰.obj i,
+  map := λ i, option.rec f 𝒰.map i,
```

Comment:
```suggestion
  obj := option.rec Y 𝒰.obj,
  map := option.rec f 𝒰.map,
```
or
```suggestion
  obj := λ i, i.rec Y 𝒰.obj,
  map := λ i, i.rec f 𝒰.map,
```
(untested)

Reply:
They both don't work. Presumably because of elaboration order issues?